src/plugins/Tus10.js

```diff
@@ -85,36 +103,35 @@ module.exports = class Tus10 extends Plugin {
 
     // Create a new tus upload
     return new Promise((resolve, reject) => {
-      const upload = new tus.Upload(file.data, {
+      var optsTus = Object.assign({}, tusDefaultOptions, this.opts)
 
-        // TODO merge this.opts or this.opts.tus here
-        metadata: file.meta,
-        resume: this.opts.resume,
-        endpoint: this.opts.endpoint,
+      optsTus.onError = (err) => {
+        this.core.log(err)
+        this.core.emitter.emit('core:upload-error', file.id, err)
+        reject('Failed because: ' + err)
+      }
 
-        onError: (err) => {
-          this.core.log(err)
-          this.core.emitter.emit('core:upload-error', file.id, err)
-          reject('Failed because: ' + err)
-        },
-        onProgress: (bytesUploaded, bytesTotal) => {
-          this.core.emitter.emit('core:upload-progress', {
-            uploader: this,
-            id: file.id,
-            bytesUploaded: bytesUploaded,
-            bytesTotal: bytesTotal
-          })
-        },
-        onSuccess: () => {
-          this.core.emitter.emit('core:upload-success', file.id, upload, upload.url)
+      optsTus.onProgress = (bytesUploaded, bytesTotal) => {
+        this.core.emitter.emit('core:upload-progress', {
+          uploader: this,
+          id: file.id,
+          bytesUploaded: bytesUploaded,
+          bytesTotal: bytesTotal
+        })
+      }
 
-          if (upload.url) {
-            this.core.log(`Download ${upload.file.name} from ${upload.url}`)
-          }
+      optsTus.onSuccess = () => {
+        this.core.emitter.emit('core:upload-success', file.id, upload, upload.url)
 
-          resolve(upload)
+        if (upload.url) {
+          this.core.log('Download ' + upload.file.name + ' from ' + upload.url)
         }
-      })
+
+        resolve(upload)
+      }
+      optsTus.metadata = file.meta
```

Comment:
Maybe we can throw an error if the user specified a `metadata` option too, since it would get overridden here? The error message could tell the user to use the `MetaData` plugin instead.